subprojects/distributions/src/integTest/groovy/org/gradle/TestDistributionIntegrationTest.groovy

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle
+
+import com.google.common.collect.Iterables
+import org.gradle.test.fixtures.file.TestFile
+
+import java.nio.file.Paths
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+
+class TestDistributionIntegrationTest extends  DistributionIntegrationSpec {
```

Comment:
Since this test (and the other distribution tests) rely on all of the built distributions (src, bin, all, test-bin), shouldn't we make the distributions project integTests all have these files as inputs? I can't quite imagine the twisted steps to take, but it would be good to cover.